cinema_project/base_app/jobs.py

```diff
@@ -4,14 +4,18 @@
 from apscheduler.schedulers.background import BackgroundScheduler
 
 
-def remove_unconfirmed_reservations() -> None:
-    Reservation.objects.filter(
+def cancel_unconfirmed_reservations() -> None:
+    reservations_to_cancel = Reservation.objects.filter(
         is_confirmed=False,
+        is_canceled=False,
         schedule__schedule_time__lte=thirty_minutes_ahead()
-    ).delete()
+    )
+    for reservation in reservations_to_cancel:
+        reservation.is_canceled = True
+        reservation.save()
```

Comment:
Here, we could use .filter(...).update(is_canceled=True) to skip the loop.

Reply:
Cool this is very nice, was looking for smth like this but didn't find it!